src/app/codegen-data-model-provider/CodegenDataModelProvider.cpp

```diff
@@ -396,21 +411,55 @@ bool CodegenDataModelProvider::EndpointExists(EndpointId endpoint)
     return (emberAfIndexFromEndpoint(endpoint) != kEmberInvalidEndpointIndex);
 }
 
-EndpointId CodegenDataModelProvider::FirstEndpoint()
+std::optional<DataModel::EndpointInfo> CodegenDataModelProvider::GetEndpointInfo(EndpointId endpoint)
+{
+    std::optional<unsigned> endpoint_idx = TryFindEndpointIndex(endpoint);
+    if (endpoint_idx.has_value())
+    {
+        return GetEndpointInfoAtIndex(static_cast<uint16_t>(*endpoint_idx));
+    }
+    return std::nullopt;
+}
+
+std::optional<DataModel::EndpointInfo> CodegenDataModelProvider::GetEndpointInfoAtIndex(uint16_t endpointIndex)
```

Comment:
This method does not seem to use any class members. please move it to a anonymous namespace function instead (do not make it a member method). We use member methods if something is required/updated within the class (usually a hint usage and update)

Reply:
Done, moved it to the anonymous namespace